src/consensus/dkg.rs

```diff
@@ -35,8 +34,9 @@ impl DkgKey {
         // Calculate the hash without involving serialization to avoid having to return `Result`.
         let mut hasher = Sha3::v256();
 
-        for name in elders_info.elders.keys() {
-            hasher.update(&name.0);
+        for peer in elders_info.elders.values() {
+            hasher.update(&peer.name().0);
+            hasher.update(&[peer.age()]);
```

Comment:
Again I feel name+age needs to be immutable if that helps.

Reply:
The reason for this is that during the startup phase, we increase ages on every churn event directly. So currently name+age is **not** immutable. But I think we should change that because it complicates things. We might use relocation instead, but we need to be careful to not trigger relocation loop (a node is relocated and as it re-joins, it triggers more relocations, etc...). I'd prefer to tackle this issue as a separate task.

Reply:
But we still have the issue of same participants with different ages will have same dkg_keys ?

Reply:
Because the `DkgKey` is different if the ages are different, it will start a new dkg session even if we have one for the same participants already running.